oxidation/crates/parsec_api_types/src/error.rs

```diff
@@ -0,0 +1,32 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use parsec_api_crypto::CryptoError;
+use thiserror::Error;
+
+use crate::DeviceID;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum DataError {
+    #[error("Invalid compression")]
+    Compression,
+
+    #[error("{exc}")]
+    Crypto { exc: CryptoError },
+
+    #[error("Invalid serialization")]
+    Serialization,
+
+    #[error("Invalid author: expected `{expected}`, got `None`")]
+    Validation1 { expected: DeviceID },
+
+    #[error("Invalid author: expected `{expected}`, got `{got}`")]
+    Validation2 { expected: DeviceID, got: DeviceID },
```

Comment:
`Validation2` is almost identical to `Validation1`.
Can those variant be merged together by using `Option` ?

Reply:
Good catch !

Reply:
The downside is adding Some around the id in the python error, which is a bit weird

Reply:
The change imply:

`DataError(f"Invalid author: expected `{repr_expected_author}`, got `{data.author}`")` 

---->

```Python
if data.author is None:
    `DataError(f"Invalid author: expected `{repr_expected_author}`, got `{data.author}`")` 
else:
    `DataError(f"Invalid author: expected `{repr_expected_author}`, got `Some({data.author})`")` 
```

Reply:
Another solution is to type `got` as `String`

Reply:
My bad, we can access self data